src/Dropdown/Dropdown.jsx

```diff
@@ -1,50 +1,408 @@
-import React from 'react'
+import React, {
+  Children,
+  cloneElement,
+  isValidElement,
+  useState,
+  useLayoutEffect,
+  useRef,
+  useMemo
+} from 'react'
 import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import css from '@styled-system/css'
+import { Flex } from '@ivoryio/kogaio'
 
-import { Dropdown as KogaioDropdown } from '@ivoryio/kogaio'
+import { Typography, typography } from '../Typography'
+import { ErrorMessage } from '../Input'
+import Option from './Option'
+import Heading from './Heading'
 
-const Dropdown = ({ children, ...props }) => (
-  <KogaioDropdown {...props}>{children}</KogaioDropdown>
+const Dropdown = ({
+  disabled,
+  multiple,
+  error,
+  label,
+  value,
+  id,
+  required,
+  noBottomSpace,
+  placeholder,
+  valueRender,
+  itemSelected,
+  itemFilter,
+  searchPlaceholder,
+  onChange,
+  children,
+  ...props
+}) => {
+  const [isCollapsed, setCollapsed] = useState(true)
+  const [isSwipe, setSwipe] = useState(false)
+  const [search, setSearch] = useState('')
+
+  const wrapper = useRef()
+  const searchField = useRef()
+
+  useLayoutEffect(() => {
+    const clickOutside = event => {
+      if (!wrapper.current.contains(event.target)) {
+        setCollapsed(true)
+      }
+    }
+    window.addEventListener('click', clickOutside, { passive: true })
+    return () => {
+      window.removeEventListener('click', clickOutside, { passive: true })
+    }
+  }, [])
+
+  const isEmpty = useMemo(
+    () => (value ? (multiple ? value.length === 0 : !value) : true),
+    [value, multiple]
+  )
+
+  const hasSearch = useMemo(
+    () => Array.isArray(children) && children.length > 40,
+    [children]
+  )
```

Comment:
Possible improvement: Un `searchThreshold = 40` in props  - in cazul in care vrei search sa ii faci override sa ai search pe < 40 items

Reply:
Yep, just a letfover from coding...